AutoRaise.mm

```diff
@@ -475,48 +598,22 @@ CGEventRef eventTapHandler(CGEventTapProxy proxy, CGEventType type, CGEventRef e
 }
 
 #define POLLING_MS 20
+#define VERSION "1.8"
 int main(int argc, const char * argv[]) {
     @autoreleasepool {
-        if (argc >= 3) {
-            NSUserDefaults * standardDefaults = [NSUserDefaults standardUserDefaults];
-            delayCount = abs((int) [standardDefaults integerForKey: @"delay"]);
-            warpMouse = argc != 3 ;
-            if (argc >= 7) {
-                warpX = [standardDefaults floatForKey: @"warpX"];
-                warpY = [standardDefaults floatForKey: @"warpY"];
-                if (argc >= 8) {
-                    cursorScale = [standardDefaults floatForKey: @"scale"];
-                }
-            }
-        } else {
-            NSString * home = NSHomeDirectory();
-            NSFileHandle * delayFile = [NSFileHandle fileHandleForReadingAtPath:
-                [NSString stringWithFormat: @"%@/AutoRaise.delay", home]];
-            if (delayFile) {
-                delayCount = abs([[[NSString alloc] initWithData:
-                    [delayFile readDataOfLength: 2] encoding:
-                    NSUTF8StringEncoding] intValue]);
-                [delayFile closeFile];
-            }
-            NSFileHandle * warpFile = [NSFileHandle fileHandleForReadingAtPath:
-                [NSString stringWithFormat: @"%@/AutoRaise.warp", home]];
-            if (warpFile) {
-                warpMouse = true;
-                NSString * line = [[NSString alloc] initWithData:
-                    [warpFile readDataOfLength:11] encoding:
-                    NSUTF8StringEncoding];
-                NSArray * components = [line componentsSeparatedByString: @" "];
-                if (components.count >= 1) { warpX = [[components objectAtIndex:0] floatValue]; }
-                if (components.count >= 2) { warpY = [[components objectAtIndex:1] floatValue]; }
-                if (components.count >= 3) { cursorScale = [[components objectAtIndex:2] floatValue]; }
-                [warpFile closeFile];
-            }
-        }
-        if (!delayCount) { delayCount = 2; }
-        if (!cursorScale) { cursorScale = 2; }
-
-        printf("\nBy sbmpost(c) 2021, usage:\nAutoRaise -delay <1=%dms> [-warpX <0.5> -warpY <0.5> -scale <2.5>]"
-               "\nv1.8 started with %d ms delay%s", POLLING_MS, delayCount*POLLING_MS, warpMouse ? ", " : "\n");
+        printf("\nv%s by sbmpost(c) 2021, usage:\nAutoRaise -delay <1=%dms> [-warpX <0.5> -warpY <0.5> -scale <2.0>]", VERSION, POLLING_MS);
+        
+        ConfigClass * config = [[ConfigClass alloc]init];
+        [config readCommandLineConfig: argc];
```

Comment:
probably better named: "readConfig", as it could also read from the config from a file.

Reply:
Sure!